services/rule_engine/src/main/resources/rules/smith/Heartbeat.drl

```diff
@@ -10,22 +10,26 @@ import com.mindsmiths.ruleEngine.util.Agents;
 import com.mindsmiths.gsheetsAdapter.GSheetsAdapterAPI
 import com.mindsmiths.gsheetsAdapter.reply.Spreadsheet
 
+// Connecting to google spreadsheet
 rule "Get Google Sheets"
     when
         Heartbeat(now: timestamp) from entry-point "signals"
-        agent: Smith(/*lastTableUpdate == null || lastTableUpdate before[30s] now*/)
+        agent: Smith(lastTableUpdate == null || lastTableUpdate before[30s] now)
     then
         modify(agent){
             setLastTableUpdate(now)
         };
         GSheetsAdapterAPI.getSpreadsheet();
 end
 
+// Rule for processing the sheet
 rule "Process Sheet"
     when
         spreadsheet: Spreadsheet() from entry-point "signals"
         agent: Smith()
     then
         agent.processSheet(spreadsheet);
+        //agent.getReview("karlo.visnjovski@gmail","6","oke je");
+        agent.newAgent();
```

Comment:
It would be better if you called this method from `processSheet` method.

Reply:
This method is here also just for testing. Most of it with updating sheets is. That function will be worked on by Lovro and called from the last submit button and not from process sheet.